pandas/core/indexes/multi.py

```diff
@@ -799,9 +799,11 @@ def values(self):
             box = hasattr(lev, '_box_values')
             # Try to minimize boxing.
             if box and len(lev) > len(lab):
-                taken = lev._box_values(algos.take_1d(lev._values, lab))
+                taken = lev._box_values(algos.take_1d(lev._values,
+                                                      lab))
```

Comment:
don't the algos functions always assume ndarray?

Reply:
Yes, fixed (the only one that should be `._values` is the one on line 809 below).